src/Microsoft.AspNetCore.Mvc.Abstractions/ModelBinding/ModelBindingContextDisposable.cs

```diff
@@ -0,0 +1,34 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+
+namespace Microsoft.AspNetCore.Mvc.ModelBinding
+{
+    /// <summary>
+    /// Return value of <see cref="IModelBindingContext.PushContext"/>. Should be disposed
+    /// by caller when child binding context state should be popped off of 
+    /// the <see cref="IModelBindingContext"/>.
+    /// </summary>
+    public struct ModelBindingContextDisposable : IDisposable
```

Comment:
We shouldn't need this class at all. Model binding system intentionally catches almost no `Exception`s except those which occur inside an `InputFormatter` or while performing a conversion. So we can unwind stack frames without worrying about the `ModelBindingContext` state. Just have `PushContext()` return `void` and change call sites to use `PopContext()` explicitly.